lightning/src/ln/functional_tests.rs

```diff
@@ -9628,6 +9629,202 @@ pub fn test_remove_expired_inbound_unfunded_channels() {
+
+	nodes[0]
+		.node
+		.force_close_broadcasting_latest_txn(&channel_id, &node_b_id, "manual close".to_owned())
+		.unwrap();
+	check_added_monitors!(&nodes[0], 1);
+	check_added_monitors!(&nodes[1], 0);
+
+	assert!(nodes[0].tx_broadcaster.txn_broadcast().is_empty());
+	nodes[0].node.get_and_clear_pending_msg_events();
+	nodes[1].node.get_and_clear_pending_msg_events();
+	let events = nodes[0].node.get_and_clear_pending_events();
+	assert_eq!(events.len(), 1);
+	match &events[0] {
+		Event::ChannelClosed {
+			reason: ClosureReason::HolderForceClosed { broadcasted_latest_txn, message },
+			counterparty_node_id: Some(id),
+			..
+		} => {
+			assert_eq!(*broadcasted_latest_txn, Some(true));
+			assert_eq!(message, "manual close");
+			assert_eq!(id, &node_b_id);
+		},
+		_ => panic!("Unexpected event"),
+	}
+	nodes[1].node.get_and_clear_pending_events();
+
+	let monitor_events = nodes[0].chain_monitor.chain_monitor.get_and_clear_pending_events();
+	assert!(monitor_events.is_empty());
+
+	confirm_transaction(&nodes[0], &funding_tx);
+	confirm_transaction(&nodes[1], &funding_tx);
+	nodes[0].node.get_and_clear_pending_msg_events();
+	nodes[1].node.get_and_clear_pending_msg_events();
+
+	{
+		let monitor = get_monitor!(&nodes[0], channel_id);
+		// manual override
+		monitor.broadcast_latest_holder_commitment_txn(
+			&nodes[0].tx_broadcaster,
+			&nodes[0].fee_estimator,
+			&nodes[0].logger,
+		);
+	}
+	let funding_txid = funding_tx.compute_txid();
+	let broadcasts = nodes[0].tx_broadcaster.txn_broadcast();
+	assert!(!broadcasts.is_empty());
+	let commitment_tx = broadcasts
+		.iter()
+		.find(|tx| {
+			tx.input.iter().any(|input| {
+				input.previous_output.txid == funding_txid
+					&& input.previous_output.vout == u32::from(funding_outpoint.index)
+			})
+		})
+		.expect("commitment transaction not broadcast");
+	check_spends!(commitment_tx, funding_tx);
+	assert_eq!(commitment_tx.input.len(), 1);
+	let commitment_input = &commitment_tx.input[0];
+	assert_eq!(commitment_input.previous_output.txid, funding_txid);
+	assert_eq!(commitment_input.previous_output.vout, u32::from(funding_outpoint.index));
+
+	let monitor_events = nodes[0].chain_monitor.chain_monitor.get_and_clear_pending_events();
+	assert!(monitor_events.iter().all(|event| !matches!(event, Event::BumpTransaction(_))));
+	assert!(nodes[0].node.get_and_clear_pending_events().is_empty());
+	assert!(nodes[0].node.get_and_clear_pending_msg_events().is_empty());
+}
+
+#[test]
+fn test_manual_broadcast_detects_funding_and_broadcasts_on_timeout() {
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	let node_b_id = nodes[1].node.get_our_node_id();
+
+	let (channel_id, funding_tx, funding_outpoint) =
+		create_channel_manual_funding(&nodes, 0, 1, 100_000, 10_000);
+
+	let funding_msgs =
```

Comment:
It would be nice to have the same test (for the HTLC timeout pipeline) that checks that the commitment tx only gets broadcasted after the funding confirms.